src/mutate/dropStructureChange.ts

```diff
@@ -62,19 +66,59 @@ export const dropOrphanChanges = (newTr: Transaction, dropDataTracked?: boolean)
     ) {
       changesIds.add(c.dataTracked.moveNodeId)
     }
+    if (c.dataTracked.operation === CHANGE_OPERATION.node_split) {
+      changesIds.add(c.dataTracked.id)
+    }
+    if (c.dataTracked.operation === CHANGE_OPERATION.reference) {
+      referenceIds.add(c.dataTracked.referenceId)
+    }
   })
 
-  if (!shadowIds.size && !changesIds.size) {
+  if (!shadowIds.size && !referenceIds.size && !changesIds.size) {
     return
   }
 
   changeSet.changes.forEach((c) => {
+    // remove reference if it's not pointing to any change
+    if (
+      c.dataTracked.operation === CHANGE_OPERATION.reference &&
+      !changesIds.has(c.dataTracked.referenceId)
+    ) {
+      const node = newTr.doc.nodeAt(c.from)
+      const dataTracked = node && (getBlockInlineTrackedData(node) || []).filter((d) => d.id !== c.id)
+      newTr.setNodeMarkup(c.from, undefined, { ...node?.attrs, dataTracked })
+    }
+    if (
+      c.type === 'node-change' &&
+      c.dataTracked.operation === CHANGE_OPERATION.node_split &&
+      !referenceIds.has(c.id)
+    ) {
+      const { id, ...attrs } = c.dataTracked
+      newTr.replaceWith(
```

Comment:
maybe we should use setNodeAttribute?

Reply:
I replace the node as we add insert to all of the children also